data/common.yaml

```diff
@@ -20,6 +23,13 @@ mcollective::server_config:
   plugin.yaml: "/etc/puppetlabs/mcollective/generated-facts.yaml"
   registerinterval: 10
   registration: "Choria"
+  logfile: "/var/log/puppetlabs/mcollective.log"
```

Comment:
The params class pattern is ancient and deprecated by data in modules.  This module is built around the new approach.  So saying it should be like a params.pp based class isn't really an option.

Maybe you're new to this approach but see the official docs taking this approach https://docs.puppet.com/puppet/4.3/lookup_quick_module.html and the ntp module https://github.com/puppetlabs/puppetlabs-ntp/blob/master/manifests/init.pp

Without common options, you're again left with duplicating data.

I'll likely merge this PR as is and pull out these params to params later.  Will review 'morrow